updating documentation to reflect any and all tags

RinkiyaKeDad opened the pull request:
Signed-off-by: RinkiyaKeDad <arshsharma461@gmail.com>

chipzoller commented on content/en/docs/Writing policies/match-exclude.md:
```diff
@@ -8,12 +8,24 @@ The `match` and `exclude` filters control which resources policies are applied t
 
 The `match` and `exclude` clauses have the same structure and can each contain the following elements:
 
+* `any`: specify [resource filters](#resource-filters) on which Kyverno will perform the logical **OR** operation while choosing resources
```
If the limitation exists as discussed yesterday whereby `any` and `all` are mutually exclusive with only one of either possible then this needs to be stated here.

chipzoller commented on content/en/docs/Writing policies/match-exclude.md:
```diff
@@ -136,33 +166,34 @@ spec:
-        # Optional clusterroles to be matched
-        clusterRoles: 
-        - cluster-admin
+                - {key: tier, operator: In, values: [database]}
```
Check indentation and spacing here, please.

chipzoller reviewed (commented):
This is obviously new functionality with a schema change. You've gone and updated the entirety of the match-exclude page, which raises two questions in the docs:

1. Are you going to now update *ALL* examples across the docs as well for consistency?
2. Should you perhaps not update all the examples in even this page and instead show a couple examples?

Seems like maybe 2 might be better unless you're willing to spend much more time with updating examples so it can be merged into the 1.4.2 branch.

RinkiyaKeDad commented:
Thanks for the review @chipzoller! I've updated the PR to mention that only one of `any` or `all` can be specifed. Reverted back to the original examples and just let two remain which show the new syntax. If you have any better examples in mind please let me know and I'll update the PR.

chipzoller commented on content/en/docs/Writing policies/match-exclude.md:
```diff
@@ -6,7 +6,7 @@ weight: 2
 The `match` and `exclude` filters control which resources policies are applied to.
 
-The `match` and `exclude` clauses have the same structure and can each contain the following elements:
+The `match` and `exclude` clauses have the same structure and can each contain **only one** the two elements:
```
"...of the two elements:"

chipzoller reviewed (approved):
1.4.2

JimBugwadia commented:
@RinkiyaKeDad @chipzoller - is this ready to be merged?

RinkiyaKeDad commented:
Yup @chipzoller has already approved the changes so I guess we're good to go.

realshuting commented on content/en/docs/Writing policies/match-exclude.md:
```diff
@@ -6,14 +6,26 @@ weight: 2
+Specifying resource filters directly under match and exclude has been marked for deprecation and will be removed in a future release. It is highly recommended you specify them under `any` or `all` blocks.
+{{% /alert %}}
+
+At least one element must be specified in a `match.(any/all).resources.kinds` or `exclude` block. The `kind` attribute is mandatory when working with the `resources` element. Wildcards (`*`) are currently not supported in the `match.(any/all).resources.kinds` field.
```
>The `kind` attribute is mandatory when working with the `resources` element.

@RinkiyaKeDad - is the above statement still valid? I thought we marked `match.resources` as optional and only require `kinds` if `match.resources` is used.

Quoting the comment from @windowsrefund:
>Also, the docs on https://kyverno.io/docs/writing-policies/match-exclude/#exclude-kube-system-namespace now say "The kind attribute is mandatory when working with the resources element". That's really confusing because one would think if that's true, each resources element under exclude.any would need to repeat the list of kinds in match.any.resources
